src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -54,7 +58,8 @@ internal partial class SolutionState
             ImmutableDictionary<string, ImmutableArray<DocumentId>> linkedFilesMap,
             ProjectDependencyGraph dependencyGraph,
             VersionStamp version,
-            Lazy<VersionStamp> lazyLatestProjectVersion)
+            Lazy<VersionStamp> lazyLatestProjectVersion,
+            ValueSource<SolutionStateChecksums> lazyChecksums)
```

Comment:
this is just plumbing. I am working on a PR which does checksum update more grandular. right now, all constructor of States reset lazyChecksums to be new AsyncLazy.